packages/typescript/src/options/validate.ts

```diff
@@ -65,8 +65,13 @@ export function validatePaths(
             `@rollup/plugin-typescript: Path of Typescript compiler option '${dirProperty}' must be located inside Rollup 'dir' option.`
           );
         }
+      } else if(dirProperty === 'outDir') {
+        const fromTsDirToRollup = relative(compilerOptions[dirProperty],outputDir);
+        if (fromTsDirToRollup.startsWith('..')) {
+            context.error(`@rollup/plugin-typescript: Path of Typescript compiler option '${dirProperty}' must be located inside the same directory as the Rollup 'file' option.`);
```

Comment:
since we've added a new condition here, please add a test where there _is_ an error produced. that will help prevent regressions in the future.